### PR TITLE
WIP: Dnn norm perf tests

### PR DESCRIPTION
**Pull request by ziyuanLi-alex**
## Summary

  This PR improves performance and accuracy coverage for DNN normalization layers.

  - Add RMSNormalization and MVN performance benchmarks.
  - Complete perf coverage for:
      - LayerNormalization
      - RMSNormalization
      - MVN
      - InstanceNormalization
      - GroupNormalization

  - Cover both channel-local and across_channels MVN configurations.
  - Add accuracy tests using an independent double-precision reference implementation.

  ## Commits

  - perf: add RMSNorm and MVN benchmarks
  - test: add accuracy coverage for DNN normalization layers

  ## Testing

  Built locally with GCC 11.4.0 in Release mode:

  cmake --build build_native --target opencv_test_dnn -j8

  Executed:

  ./build_native/bin/opencv_test_dnn \
    --gtest_filter="DNN_FastNorm.*"

  Result:

  [ PASSED ] 5 tests

  The following accuracy tests passed:

  - DNN_FastNorm.LayerNorm
  - DNN_FastNorm.RMSNorm
  - DNN_FastNorm.MVN
  - DNN_FastNorm.InstanceNorm
  - DNN_FastNorm.GroupNorm

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
> This PR improves performance and accuracy coverage for DNN normalization layers.

There are only new tests in the PR, but no optimization. Added WIP into PR name till it's not complete.

**Comment by abhishek-gola** [comment]
@ziyuanLi-alex Gentle Reminder!!!

**Comment by ziyuanLi-alex** [comment]
Added a commit that implements SIMD for existing normalization kernels.

Note: accuracy tests for five kernels (LayerNorm, RMSNorm, MVN, InstanceNorm, GroupNorm) are newly implemented in this PR.  Resutls are compared against independent double-precision reference calculations, generally using absolute tolerance 1e-5 and relative tolerance 1e-4.
And performace tests for RMSNorm, MVN, MVN across Channels are newly added in this PR. 

### Accuracy Tests

```
CTEST_FULL_OUTPUT
OpenCV version: 5.1.0-dev
OpenCV VCS version: 5.0.0-214-ge32210cf93
Build type: Release
Compiler: /opt/toolchains/spacemit-toolchain-linux-glibc-x86_64-v1.2.4/bin/riscv64-unknown-linux-gnu-clang++  (ver 21.1.8)
Algorithm hint: ALGO_HINT_ACCURATE
HAL: YES (RVV HAL (ver 0.0.1))
Parallel framework: pthreads (nthreads=8)
CPU features: RVV
TEST: Skip tests with tags: 'mem_6gb', 'verylong', 'dnn_skip_opencv_backend', 'dnn_skip_cpu', 'dnn_skip_cpu_fp16', 'dnn_skip_ocl', 'dnn_skip_ocl_fp16', 'dnn_skip_onnx_conformance', 'dnn_skip_parser'
Note: Google Test filter = DNN_FastNorm.*
[==========] Running 5 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 5 tests from DNN_FastNorm
[ RUN      ] DNN_FastNorm.LayerNorm
[       OK ] DNN_FastNorm.LayerNorm (2 ms)
[ RUN      ] DNN_FastNorm.RMSNorm
[       OK ] DNN_FastNorm.RMSNorm (1 ms)
[ RUN      ] DNN_FastNorm.MVN
[       OK ] DNN_FastNorm.MVN (0 ms)
[ RUN      ] DNN_FastNorm.InstanceNorm
[       OK ] DNN_FastNorm.InstanceNorm (1 ms)
[ RUN      ] DNN_FastNorm.GroupNorm
[       OK ] DNN_FastNorm.GroupNorm (0 ms)
[----------] 5 tests from DNN_FastNorm (4 ms total)

[----------] Global test environment tear-down
[==========] 5 tests from 1 test case ran. (5 ms total)
[  PASSED  ] 5 tests.
```

### Performace


Name of Test | normbenchmarkk1 | normperfk1patch | normperfk1patchvsnormbenchmarkk1(x-factor)
-- | -- | -- | --
GroupNorm::Layer::OCV/CPU | 13.439 | 9.364 | 1.44
InstanceNorm::Layer::OCV/CPU | 8.198 | 8.090 | 1.01
LayerNorm::Layer::OCV/CPU | 0.063 | 0.057 | 1.10
MVN::Layer::OCV/CPU | 8.197 | 8.082 | 1.01
MVNAcrossChannels::Layer_MVN::OCV/CPU | 13.555 | 9.222 | 1.47
RMSNorm::Layer::OCV/CPU | 0.054 | 0.052 | 1.04

**Comment by ziyuanLi-alex** [comment]
The Test_TensorFlow_layers.batch_norm_11/0 is failing on various platforms. This is caused by rewriting the expression of  BatchNorm which is mathematically the same but creates a computation variance. 

I will apply the second patch shortly, after I run another dnn accuracy test on my side.


### Logs

Replicating the issue on my k1: 
``` 
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from Test_TensorFlow_layers
[ RUN      ] Test_TensorFlow_layers.batch_norm_11/0, where GetParam() = OCV/CPU
/home/zyuanli/cvbenchmark/opencv/modules/dnn/test/test_common.impl.hpp:109: Failure
Expected: (normL1) <= (l1), actual: 0.000485437 vs 1e-05
  |ref| = 1.7365455627441406
/home/zyuanli/cvbenchmark/opencv/modules/dnn/test/test_common.impl.hpp:112: Failure
Expected: (normInf) <= (lInf), actual: 0.000717603 vs 0.0001
  |ref| = 1.7365455627441406
input: [1 x 5 x 1 x 1]
[0.25558338, -0.50478905, -1.3044388, 2.1413262, 1.2469226]
ref [1 x 5 x 1 x 1]
[1.7365456, -0.22727542, -1.3859636, -0.52573568, 0.11772589]
output: [1 x 5 x 1 x 1]
[1.7362134, -0.22748862, -1.3864141, -0.52502191, 0.11844349]
[  FAILED  ] Test_TensorFlow_layers.batch_norm_11/0, where GetParam() = OCV/CPU (6 ms)
[----------] 1 test from Test_TensorFlow_layers (6 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (7 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Test_TensorFlow_layers.batch_norm_11/0, where GetParam() = OCV/CPU

 1 FAILED TEST
```

After patch:
```
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from Test_TensorFlow_layers
[ RUN      ] Test_TensorFlow_layers.batch_norm_11/0, where GetParam() = OCV/CPU
[       OK ] Test_TensorFlow_layers.batch_norm_11/0 (8 ms)
[----------] 1 test from Test_TensorFlow_layers (9 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (10 ms total)
[  PASSED  ] 1 test.

```

### Performance

Name of Test | normbenchmarkk1 | normperfk1patch2 | normperfk1patch2vsnormbenchmarkk1(x-factor)
-- | -- | -- | --
GroupNorm::Layer::OCV/CPU | 13.439 | 9.297 | 1.45
InstanceNorm::Layer::OCV/CPU | 8.198 | 8.086 | 1.01
LayerNorm::Layer::OCV/CPU | 0.063 | 0.054 | 1.16
MVN::Layer::OCV/CPU | 8.197 | 8.037 | 1.02
MVNAcrossChannels::Layer_MVN::OCV/CPU | 13.555 | 11.428 | 1.19
RMSNorm::Layer::OCV/CPU | 0.054 | 0.051 | 1.07

**Comment by asmorkalov** [comment]
@ziyuanLi-alex Is the PR ready for review?

**Comment by ziyuanLi-alex** [comment]
Yes, the patches are included in the #1034643 commit and I believe the CI also proves it for the failures are likely caused by network issues on certain systems

**Comment by asmorkalov** [comment]
Another PR with optimizations has been merged to 5.x. Please reopen the PR, if you have extra optimizations versus we have in 5.x now.
